ocs_ci/ocs/longevity.py

```diff
@@ -0,0 +1,711 @@
+
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import CommandFailed, UnsupportedWorkloadError
+from ocs_ci.utility.utils import ocsci_log_path
+from ocs_ci.ocs.scale_noobaa_lib import (
+    construct_obc_creation_yaml_bulk_for_kube_job,
+    check_all_obc_reached_bound_state_in_kube_job,
+)
+from ocs_ci.utility import templating
+from ocs_ci.ocs.scale_lib import (
+    construct_pvc_creation_yaml_bulk_for_kube_job,
+    check_all_pvc_reached_bound_state_in_kube_job,
+    check_all_pod_reached_running_state_in_kube_job,
+)
+from ocs_ci.ocs.resources.objectconfigfile import ObjectConfFile
+from ocs_ci.ocs.pgsql import Postgresql
+from ocs_ci.ocs.couchbase import CouchBase
+from ocs_ci.ocs.cosbench import Cosbench
+from ocs_ci.helpers.helpers import create_unique_resource_name
+from concurrent.futures import ThreadPoolExecutor
+from ocs_ci.ocs.resources.pvc import get_pvc_objs
+from ocs_ci.deployment.deployment import setup_persistent_monitoring
+from ocs_ci.ocs.registry import (
+    change_registry_backend_to_ocs,
+    check_if_registry_stack_exists,
+)
+from ocs_ci.utility.deployment_openshift_logging import install_logging
+from ocs_ci.ocs.monitoring import check_if_monitoring_stack_exists
+
+log = logging.getLogger(__name__)
+
+
+supported_app_workloads = ["pgsql", "couchbase", "cosbench"]
+supported_ocp_workloads = ["logging", "monitoring", "registry"]
+
+
+class Longevity(object):
+
+    """
+    This class consists of the library functions and params required for Longevity testing
+    """
+
+    def __init__(self):
+        """
+        Initializer function
+        """
+        lcl = locals()
+        self.tmp_path = pathlib.Path(ocsci_log_path())
+        self.pvc_size = None
+        self.pvc_count = None
+        self.pod_count = None
+        self.num_of_obcs = None
+
+    def construct_stage_builder_bulk_pvc_creation_yaml(self, num_of_pvcs, pvc_size):
+        """
+        This function constructs pvc.yamls to create bulk of pvc's using kube_jobs.
+        It constructs yamls with the specified number of PVCs of each supported
+        type and access mode.
+
+        Eg: num_of_pvcs = 30
+        The function creates a total of 120 PVCs of below types and access modes
+        RBD-Filesystemvolume  -> 30 RWO PVCs
+        CEPHFS                -> 30 RWO PVCs, 30 RWX PVCs
+        RBD-Block             -> 30 RWX PVCs
+
+        Args:
+            num_of_pvcs(int): Bulk PVC count
+            pvc_size (str): size of all pvcs to be created with Gi suffix (e.g. 10Gi).
+            If None, random size pvc will be created
+
+        Returns:
+             pvc_dict_list (list): List of all PVC.yaml dicts
+
+        """
+        pvc_dict_list = []
+        # Construct bulk pvc creation yaml for ceph-rbd with access mode RWO
+        log.info(
+            "Constructing bulk pvc creation yaml for ceph-rbd with access mode RWO"
+        )
+        pvc_dict_list1 = construct_pvc_creation_yaml_bulk_for_kube_job(
```

Comment:
instead of pvc_dict_list1, it would be better to have rbd_rwo_pvc_list and similarly in all the other places below

pvc_dict_list2, 3 and 4 can be replaced in similar fashion

Reply:
Done